src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2555,6 +2555,259 @@ class StubGenerator: public StubCodeGenerator {
+    const Register key                 = c_rarg2;
+    const Register counter             = c_rarg3;
+    const Register input_len           = c_rarg4;
+    const Register saved_encrypted_ctr = c_rarg5;
+    const Register used_ptr            = c_rarg6;
+    const Register tmp                 = c_rarg7;
+
+    const Register keylen              = x31;
+    const Register used                = x30;
+    const Register len                 = x29;
+    const Register len32               = x28;
+    const Register vl                  = t1;
+    const Register ctr                 = t2;
+
+    const unsigned char block_size = 16;
+
+    VectorRegister working_vregs[] = {
+      v1, v2, v3, v4, v5, v6, v7, v8,
+      v9, v10, v11, v12, v13, v14, v15
+    };
+
+    const address start = __ pc();
+    __ enter();
+
+    Label L_exit;
+
+    __ lw(used, Address(used_ptr));
+    __ beqz(input_len, L_exit);
+    __ mv(len, input_len);
+
+    // Init 0b01010101... v0 mask for counter increase
+    uint64_t maskIndex = 0xaaul;
+    __ vsetvli(t0, x0, Assembler::e8, Assembler::m1);
+    __ mv(t0, maskIndex);
+    __ vmv_v_x(v0, t0);
+
+    Label L_aes128_loadkeys, L_aes192_loadkeys, L_exit_loadkeys;
+    // Compute #rounds for AES based on the length of the key array
+    __ lw(keylen, Address(key, arrayOopDesc::length_offset_in_bytes() - arrayOopDesc::base_offset_in_bytes(T_INT)));
+    __ vsetivli(x0, 4, Assembler::e32, Assembler::m1);
+    __ mv(t0, 52);
+    __ blt(keylen, t0, L_aes128_loadkeys);
+    __ beq(keylen, t0, L_aes192_loadkeys);
+
+    // Load aes keys into working_vregs according to the keylen
+    generate_aes_loadkeys(key, working_vregs, 15);
+    __ j(L_exit_loadkeys);
+
+    __ bind(L_aes192_loadkeys);
+    generate_aes_loadkeys(key, working_vregs, 13);
+    __ j(L_exit_loadkeys);
+
+    __ bind(L_aes128_loadkeys);
+    generate_aes_loadkeys(key, working_vregs, 11);
+    __ bind(L_exit_loadkeys);
+
+    Label L_next, L_encrypt_next, L_main;
+
+    // Encrypt bytes left with last encryptedCounter
+    __ bind(L_next);
+    __ mv(t0, block_size);
+    __ bge(used, t0, L_main);
+
+    __ bind(L_encrypt_next);
+    __ add(tmp, saved_encrypted_ctr, used);
+    __ lbu(t0, Address(tmp));
+    __ lbu(tmp, Address(in));
+    __ xorr(t0, t0, tmp);
+    __ sb(t0, Address(out));
+    __ addi(in, in, 1);
+    __ addi(out, out, 1);
+    __ addi(used, used, 1);
+    __ subi(len, len, 1);
+    __ beqz(len, L_exit);
+    __ j(L_next);
+
+    Label L_first_loop, L_loop, L_calculate_one_next;
+
+    __ bind(L_main);
+    // Check whether the counter increases overflow 64 bit and changes to the scalar path,
+    // most of time the counter may not increase over 64 bit,
+    // but when counter starts with a large init number, we may encounter overflow one time
+    // at most, just calculate it on the scalar path, and then go back to the large block
+    __ addi(t0, counter, 8);
+    __ ld(tmp, Address(t0));
+    __ rev8(tmp, tmp);
```

Comment:
Note that `rev8` is only available under `UseZbb`. Maybe you should use `revb/revbw` instead which considers that the availability of Zbb extension.